File: c#/src/Docmost.Sdk/Services/PagesApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class PagesApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetPageAsync(PageInfoDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CreatePageAsync(CreatePageDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdatePageAsync(UpdatePageDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeletePageAsync(DeletePageDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/delete", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> RestorePageAsync(PageIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/restore", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetRecentPagesAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/recent", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetDeletedPagesAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/trash", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetPageHistoryAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/history", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetPageHistoryInfoAsync(PageHistoryIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/history/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetSidebarPagesAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/sidebar-pages", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MovePageToSpaceAsync(MovePageToSpaceDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/move-to-space", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DuplicatePageAsync(DuplicatePageDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/duplicate", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> MovePageAsync(MovePageDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/move", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetPageBreadcrumbsAsync(PageIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("pages/breadcrumbs", request, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/pages-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { CreatePageDto, DeletePageDto, DuplicatePageDto, MovePageDto, MovePageToSpaceDto, PageHistoryIdDto, PageIdDto, PageInfoDto, UpdatePageDto } from '../models/index.js';


export class PagesApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getPage(request: PageInfoDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/info`, request);
  }

  async createPage(request: CreatePageDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/create`, request);
  }

  async updatePage(request: UpdatePageDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/update`, request);
  }

  async deletePage(request: DeletePageDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/delete`, request);
  }

  async restorePage(request: PageIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/restore`, request);
  }

  async getRecentPages(): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/recent`, undefined);
  }

  async getDeletedPages(): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/trash`, undefined);
  }

  async getPageHistory(): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/history`, undefined);
  }

  async getPageHistoryInfo(request: PageHistoryIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/history/info`, request);
  }

  async getSidebarPages(): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/sidebar-pages`, undefined);
  }

  async movePageToSpace(request: MovePageToSpaceDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/move-to-space`, request);
  }

  async duplicatePage(request: DuplicatePageDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/duplicate`, request);
  }

  async movePage(request: MovePageDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/move`, request);
  }

  async getPageBreadcrumbs(request: PageIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`pages/breadcrumbs`, request);
  }

}
